test_report/src/report.rs

```diff
@@ -0,0 +1,247 @@
+use chrono::prelude::*;
+#[cfg(feature = "ruby")]
+use magnus::{value::ReprValue, Module, Object};
+use prost_wkt_types::Timestamp;
+use proto::test_context::test_run::{TestCaseRun, TestCaseRunStatus, TestResult, UploaderMetadata};
+use std::cell::RefCell;
+#[cfg(feature = "wasm")]
+use wasm_bindgen::prelude::wasm_bindgen;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct TestReport {
+    test_result: TestResult,
+}
+
+#[cfg_attr(feature = "wasm", wasm_bindgen)]
+#[cfg_attr(feature = "ruby", magnus::wrap(class = "TestExecution"))]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct TestExecution {
+    id: Option<String>,
+    name: String,
+    classname: String,
+    file: String,
+    parent_name: String,
+    line: Option<i32>,
+    status: Status,
+    attempt_number: i32,
+    started_at: i64,
+    finished_at: i64,
+    output: String,
+}
+
+#[cfg_attr(feature = "wasm", wasm_bindgen)]
+#[cfg_attr(feature = "ruby", magnus::wrap(class = "Status"))]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Status {
+    Success,
+    Failure,
+    Skipped,
+}
+
+#[cfg(feature = "ruby")]
+impl Status {
+    fn new(status: String) -> Self {
+        match status.as_str() {
+            "success" => Status::Success,
+            "failure" => Status::Failure,
+            "skipped" => Status::Skipped,
+            _ => Status::Success,
```

Comment:
Agreed, but `magnus` makes this a real pain. We can `panic` and it would be handled upstream in ruby's case but I'm not sure how this will be handled by other integrations.